portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -49,14 +54,43 @@ public String toString() {
     public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
         String name = request.getParameter("name");
         String comment = request.getParameter("comment");
-        Comment entry = new Comment(name, comment);
 
-        allComments.add(entry);
+        Entity taskEntity = new Entity("Task");
+        taskEntity.setProperty("name", name);
+        taskEntity.setProperty("comment", comment);
+
+        datastore.put(taskEntity);
         response.sendRedirect("/index.html");
     }
 
     @Override
     public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        String input = request.getParameter("comment-number");
+        int limit = 0;
+        
+        try {
+            limit = Integer.parseInt(input);
+        } catch (NumberFormatException e) {
+            System.err.println("Could not convert to int: " + input);
+            return;
+        }
+
+        Query query = new Query("Task");
+        PreparedQuery results = datastore.prepare(query);
+        
+        ArrayList<Comment> allComments = new ArrayList<Comment>();
+        int index = 0;
+        for (Entity entity : results.asIterable()) {
+            if (index >= limit) {
```

Comment:
Is there a better way to limit the number of comments returned? Does the Query API support anything like this?

Reply:
Changed. I simplified using the FetchOptions withLimit() function.